mypyc/codegen/emitfunc.py

```diff
@@ -216,26 +246,31 @@ def visit_get_attr(self, op: GetAttr) -> None:
                 self.ctype(rtype.attr_type(op.attr)),
                 op.attr))
         else:
-            typ, decl_cl = cl.attr_details(op.attr)
-            # FIXME: We use the lib_prefixed version which is an
-            # indirect call we can't inline. We should investigate
-            # duplicating getter/setter code.
-            self.emit_line('%s = %s%s((%s *)%s); /* %s */' % (
-                dest,
-                self.emitter.get_group_prefix(decl_cl),
-                native_getter_name(decl_cl, op.attr, self.emitter.names),
-                decl_cl.struct_name(self.names),
-                obj,
-                op.attr))
+            # Otherwise, use direct or offset struct access.
+            attr_expr = self.get_attr_expr(obj, op, decl_cl)
+            if attr_rtype.is_refcounted:
+                self.emitter.emit_undefined_attr_check(
+                    attr_rtype, attr_expr, '==', unlikely=True
+                )
+                exc_class = 'PyExc_AttributeError'
+                self.emitter.emit_lines(
+                    'PyErr_SetString({}, "attribute {} of {} undefined");'.format(
+                        exc_class, repr(op.attr), repr(cl.name)),
+                    '} else {')
+                self.emitter.emit_inc_ref(attr_expr, attr_rtype)
+                self.emitter.emit_line('}')
+            self.emitter.emit_line('{} = {};'.format(dest, attr_expr))
```

Comment:
Let's do the assignment to the destination before we do the checks. I think that will actually save us a load, since the compiler won't be able to assume that the value stays the same across the incref call.